### ISSUE TITLE
License not persisting between reboots

### ISSUE THREAD

**Issue by anthropos9** [yuezk/GlobalProtect-openconnect#376]
**Describe the bug**
I bought a license and added it to my client. It worked, but after rebooting my computer it now says that I don't have a license installed.

**Expected behavior**
The license should persist between reboots until it expires.

**Logs**
```
[2024-06-13T16:38:12Z INFO  gpservice::cli] gpservice started: 2.3.1 (2024-05-22)
[2024-06-13T16:38:12Z INFO  gpservice::ws_server] WS server listening on port: 46055
[2024-06-13T16:38:12Z INFO  gpapi::process::gui_launcher] Version check passed: 2.3.1
[2024-06-13T16:38:12Z INFO  gpapi::process::gui_launcher] Launching gpgui
[2024-06-13T16:38:12Z INFO  gpgui::cli] gpgui started: 2.3.1 (2024-05-22)
[2024-06-13T16:38:12Z INFO  gpgui::app] Setting the custom openssl conf path
[2024-06-13T16:38:12Z INFO  gpgui::config::private_data] Loaded config key from keyring
[2024-06-13T16:38:12Z INFO  gpgui::app::app_initializer] App initialized
[2024-06-13T16:38:12Z INFO  gpgui::ws_connector] Connecting to WS server
[2024-06-13T16:38:12Z INFO  gpgui::ws_connector] Received ping
[2024-06-13T16:38:12Z INFO  gpgui::ws_connector] Connected to WS server
[2024-06-13T16:38:12Z INFO  gpservice::handlers] New client connected
[2024-06-13T16:38:12Z INFO  gpservice::ws_server] Sending current VPN state to new client

** (gpgui:86258): WARNING **: 09:38:12.521: webkit_settings_set_enable_offline_web_application_cache is deprecated and does nothing.
[2024-06-13T16:38:13Z INFO  gpapi::utils::window] Window raised after 1 attempts
[2024-06-13T16:38:14Z INFO  gpgui::handlers::subscription] Sending the init event to client: main
[2024-06-13T16:38:14Z INFO  gpgui::handlers::subscription] Sent the init event to client: main

** (gpgui:86258): WARNING **: 09:41:54.189: webkit_settings_set_enable_offline_web_application_cache is deprecated and does nothing.
[2024-06-13T16:41:54Z INFO  gpapi::utils::window] Window raised after 2 attempts
[2024-06-13T16:41:56Z INFO  gpgui::handlers::subscription] Sending the init event to client: settings
[2024-06-13T16:41:56Z INFO  gpgui::handlers::subscription] Sent the init event to client: settings
```

**Environment:**
 - OS: Linux Mint 21.3
 - Desktop Environment: Cinnamon 
 - Output of `ps aux | grep 'gnome-keyring\|kwalletd5' | grep -v grep`: [Required for secure store error] 
 `sdean       3637  0.0  0.0 463004  9152 ?        SLl  Jun12   0:05 /usr/bin/gnome-keyring-daemon --daemonize --login`
 - Is remote SSH? Yes

**Comment by habby1337** [comment]
same problem on linux arch, license key not persisting:

- OS: Arch Linux x86_64
- Kernel: 6.9.3-zen1-1-zen
- Desktop Environment: Plasma 6.0.5 on wayland
- not remote

**Comment by yuezk** [comment]
Hi @anthropos9, @habby1337 can you help check the content under the `~/.config/com.yuezk.gpgui` folder? Thanks.

**Comment by habby1337** [comment]
```
# .config/com.yuezk.gpgui 
> ls
04BF1B5F147E-license_data.bin  5A150DDFF0F3-license_data.bin  app_config.bin  FAF0B1918346-license_data.bin
```
what should i look for?

**Comment by yuezk** [comment]
The license file is generated. The license name is prefixed by the MAC address. Looks like your MAC address changes when restarts.

**Comment by yuezk** [comment]
I'll try to fix this.

**Comment by habby1337** [comment]
thank you very much.

just an heads up i never changed anything on the machine, is a laptop, so every component is still the same.

but while this is getting fixed, is there a way to increase the licenses or a way to use the CLI so i can send the fake HIP report and connect to the portal (pretty much the same configs as the gui settings) so i can keep working?

**Comment by yuezk** [comment]
The license can be managed on https://app.lemonsqueezy.com/my-orders, feel free to deactivate the unused activations.

The CLI version also supports passing the HIP script, via the `--csd-wrapper` parameter. You can see `gpclient help connect` for details.

**Comment by anthropos9** [comment]
This is what's in my `~/.config/com.yuezk.gpgui` directory
```
 sdean@sitecrafting$  ls
total 12K
-rw------- 1 sdean  107 Jun 12 11:14 00E04C68169D-license_data.bin
-rw------- 1 sdean  107 Jun 12 13:05 8C1D96D5D414-license_data.bin
-rw-r--r-- 1 sdean 1.1K Jun 12 13:05 app_config.bin
```

**Comment by yuezk** [comment]
@anthropos9 @habby1337 This should be fixed in 2.3.2.

**Comment by anthropos9** [comment]
@yuezk my initial tests have been positive. I added the license then rebooted and it persisted. Thanks for your help.
